go/types/subtype_test.go

```diff
@@ -758,3 +758,46 @@ func TestIsValueSubtypeOf(tt *testing.T) {
 		assertFalse(v, t)
 	}
 }
+
+func TestIsValueSubtypeHasExtras(tt *testing.T) {
+	assert := assert.New(tt)
+
+	test := func(v Value, t *Type, is, he bool) {
+		isSub, hasExtra := IsValueSubtypeOfDetails(v, t)
+		assert.Equal(is, isSub, "isSub had unexpected value: %t", isSub)
+		assert.Equal(he, hasExtra, "hasExtra had unexpected value: %t", hasExtra)
+	}
+
+	test(
+		NewStruct("Struct", StructData{"x": Bool(true)}),
+		MakeStructType("Struct"),
+		true, true,
+	)
+	test(
+		NewStruct("Struct", StructData{"x": Bool(true)}),
+		MakeStructType("Struct",
+			StructField{"x", BoolType, true},
+			StructField{"y", NumberType, true},
+		),
+		true, false,
+	)
+	test(
+		NewStruct("Struct", StructData{
+			"x": Bool(true),
+			"z": Bool(true),
+		}),
+		MakeStructType("Struct",
+			StructField{"x", BoolType, true},
+			StructField{"y", NumberType, true},
+		),
+		true, true,
+	)
+	test(
+		NewStruct("Struct", StructData{"x": Bool(true)}),
+		MakeStructType("Struct",
+			StructField{"x", BoolType, false},
+			StructField{"y", NumberType, false},
+		),
+		false, false,
+	)
```

Comment:
No test for `false, true`?

Maybe structure this as in TestIsSubtypeDisallowExtraStructFields so that we can easily covefr more cases?